src/runtime/cuda/cuda_module.cc

```diff
@@ -164,11 +164,18 @@ class CUDAWrappedFunc {
   void operator()(TVMArgs args, TVMRetValue* rv, void** void_args) const {
     int device_id;
     CUDA_CALL(cudaGetDevice(&device_id));
+    ThreadWorkLoad wl = launch_param_config_.Extract(args);
+
     if (fcache_[device_id] == nullptr) {
       fcache_[device_id] = m_->GetFunc(device_id, func_name_);
+      if (wl.dyn_shmem_size >= (48 << 10)) {
```

Comment:
if dynamic memory is too large, will it pass `VerifyGPUCode` check?

Reply:
Haven't tested but yeah, it seems `VerifyGPUCode` checks the static alloc size against `max_shared_memory_per_block`, which would fail if `dyn_shmem_size >= (48 << 10)`

https://github.com/apache/tvm/blob/534205baba2a3e71bc0613824e62cb625677f3e0/src/tir/analysis/verify_gpu_code.cc#L70-L71

Reply:
Can we defer this issue later? I need this to demonstrate that a multi-stage pipeline with depth > 2 works  on a semi-realistic cuda schedule.

Reply:
Yeah let's defer this particular issue